inspirehep/modules/authors/views.py

```diff
@@ -230,21 +228,14 @@ def new():
 @login_required
 def update(recid):
     """View for INSPIRE author update form."""
-    data = {}
-    if recid:
-        try:
-            url = get_legacy_url_for_recid(recid) + '/export/xm'
-            xml = requests.get(url)
-            record_regex = re.compile(
-                r"\<record\>.*\<\/record\>", re.MULTILINE + re.DOTALL)
-            xml_content = record_regex.search(xml.content).group()
-            data = marcxml2record(xml_content)
-            convert_for_form(data)
-        except requests.exceptions.RequestException:
-            pass
-        data["control_number"] = recid
-    else:
-        return redirect(url_for("inspirehep_authors.new"))
+    try:
+        data = get_db_record('aut', recid)
+        convert_for_form(data)
+    except RecordGetterError:
+        abort(404)
```

Comment:
I just realized that we can't do this yet: many author records still don't migrate successfully, so if we were to do this we would serve a 404 instead of creating a workflow in ERROR state. So the previous logic should be moved here (but it still needs a refactoring).